### PR TITLE
Events epic

### PR DESCRIPTION
**Pull request by cgolinski**
Probably did too many things in one PR. 

Note that once the categories are clickable, the set_categories reducer will need to be updated to set the correct chosenCategory to state. The get_events action will need to fire on the same onClick.

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -18,19 +19,31 @@ class App extends Component {
...
-            handleClick={ () => setSearchKeyword(searchInput) }
-          />
-          <CategoriesList allCategories={allCategories} selectedCategories={selectedCategories} />
+        <Button
+          className='hello'
+          aria-label='hello'
+          content='hello'
+          handleClick={ () => setSearchKeyword(searchInput) }
+        />
+        <CategoriesList allCategories={allCategories} selectedCategories={selectedCategories} />
```
Comment (srhboo) [inline src/components/App.js]: the merged category routing PR will have conflicts here (this should be {childrenWithProps} which will automatically plug in CategoriesList or CategoryPage based on page route url)

File: src/components/App.js [hunk 2]
```diff
@@ -18,19 +19,31 @@ class App extends Component {
...
+        <button onClick={ () => getCategories() }>
```
Comment (epq) [inline src/components/App.js]: The button for fetching categories isn't needed anymore.

File: src/components/App.js [hunk 3]
```diff
@@ -18,19 +19,31 @@ class App extends Component {
...
+          Fetch Categories
+        </button>
+        { chosenCategory
+          ? <CategoryPage chosenCategory={chosenCategory} getEvents={getEvents} events={events}/>
+          : null
+        }
+        {console.log('chosenCategory: ', chosenCategory)}
```
Comment (epq) [inline src/components/App.js]: Remove this? Unless it's useful for debugging.

File: src/components/CategoryPage.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { EventsList } from './EventsList';
+
+export const CategoryPage = (props) => (
+  <div>
+    {console.log('props.chosenCategory in CategoryPage component: ', props.chosenCategory)}
```
Comment (epq) [inline src/components/CategoryPage.js]: Remove this too.

**Comment by srhboo** [comment]
Hey @cgolinski nice work! I left a comment on App.js -> there will be a conflict here, check the recently merged route-category App.js file and make sure {childrenWithProps} is used because that will handle the displaying CategoriesList / CategoryPage logic. The other code looks good!

File: src/components/Event.js [hunk 5]
```diff
@@ -3,14 +3,16 @@ import React from 'react';
 export const Event = event => {
   return(
     <li className='item-card'>
-      <a href={event.href}>
+      <a href={event.url}>
         <div className='item-container'>
-          <img src={event.img}/>
+{/*          <img src={event.logo.url}/>*/}
```
Comment (epq) [inline src/components/Event.js]: Do you need to set the image?

File: src/redux/epics/event.epics.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { ACTION_TYPES } from '../../constants/ActionTypes';
+import { Observable } from 'rxjs';
+import 'rxjs/add/operator/mergeMap';
+
+const EVENTBRITE_API_KEY = 'QWZWM6CDIXJRAOHDGJ7Y';
```
Comment (epq) [inline src/redux/epics/event.epics.js]: Should we put this in a constants file?

File: src/components/App.js (hunk 2 above)
Comment (auredufour) [inline src/components/App.js]: @epq I removed it in my PR, when it will be merge, it will update the App.js

File: src/components/Event.js (hunk 5 above)
Comment (auredufour) [inline src/components/Event.js]: @epq, it is a placeholder I put until I can have the data on the Event component!

Comment (cgolinski) [inline src/components/Event.js]: Removed for now.

File: src/redux/epics/event.epics.js (hunk 6 above)
Comment (cgolinski) [inline src/redux/epics/event.epics.js]: Good idea. Done.

**Comment by srhboo** [comment]
👍

**Comment by auredufour** [comment]
WOW! You did so much, good job! 👍

**Pull request merged**